nrf_security/Kconfig.legacy

```diff
@@ -328,7 +328,7 @@ comment "Cipher Selection"
 
 config MBEDTLS_CIPHER_MODE_CBC
 	bool "AES-CBC - AES Cipher Block Chaining mode"
-	default y
+	default y if !NET_L2_OPENTHREAD
```

Comment:
I don't think that nRF Security should have knowledge / dependency of OpenThread.

If defaults should be different when OpenThread is enabled, then I would say OpenThread should provide a defconfig on how security should look.

If nRF security starts having dependencies to various non-security related modules, then we are en route to hit circular dependency issues soon.

This comment applies to all `NET_L2_OPENTHREAD` references in this file.